Gitlab CICD.sublime-syntax

```diff
@@ -63,3 +63,40 @@ contexts:
     - meta_include_prototype: false
     - meta_scope: source.shell.bash.embedded
     - include: scope:source.shell.bash.folded
+
+  property:
+    - meta_prepend: true
+    - match: (!reference)\s*(\[)
+      captures:
+        1: meta.property.yaml storage.type.tag-handle.yaml
+        2: meta.sequence.flow.yaml punctuation.definition.sequence.begin.yaml
+      push: gitlab-reference-flow-sequence-body
+
+  gitlab-reference-flow-sequence-body:
+    - meta_content_scope: meta.sequence.flow.yaml
+    - match: \]
+      scope: meta.sequence.flow.yaml punctuation.definition.sequence.end.yaml
+      pop: 1
+    - match: ','
+      scope: punctuation.separator.sequence.yaml
+      push: gitlab-after-reference
+    - match: '[-\w.]+'
+      scope: constant.other.label.gitlab
+    - include: flow-pair-no-clear
+    - include: flow-node-11
+
+  gitlab-after-reference:
+    - match: \w+
+      scope: keyword.other.gitlab
+    - match: ','
+      scope: punctuation.separator.sequence.yaml
+      set: gitlab-after-reference-type
+    - match: (?=\])
+      pop: 1
+
+  gitlab-after-reference-type:
+    - match: \w+
+      scope: variable.other.constant.gitlab
+      pop: 1
```

Comment:
I suggest to not make this pop but allow matching multiple segments, i.e. also a comma and quoted scalars.

Reply:
done